docs/DeveloperGuide.md

```diff
@@ -1,13 +1,173 @@
+| Parser        | Provides functions to parse inputs read from standard input.                        |
+| Data          | Stores data and provides functions to operate on data.                              |
+| Storage       | Defines functions to save and load data.                                            |
+| Logger        | Defines functions to log the user's actions and the program's behaviour.            |
+
+### Component Interactions:
+The sequence diagram below describes the interaction between the various core components when a command is entered.
+In this example, 
+the user launches the program and enters the command `Add-Expense -n Expense -a 7.80` to add an expense with the name `Expense` and the amount `7.80`.
+The sequence diagrams referenced by the component interaction diagram can be seen [below](#component-interaction-sequence-diagrams)
+
+![Component-Interaction-On-Command-Entered](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/xzynos/tp/branch-MoneyGoWhere-Webpage/docs/diagrams/ComponentInteractionsOnCommandEntered.puml)
+
+* When the user launches the program, `MoneyGoWhere` creates an instance of `UserInterface`.
+* `MoneyGoWhere` calls `UserInterface#run()` to start the interface between the program and the user.
+* `UserInterface#run()` will execute continuously in a loop until the user enters the command `Bye`.
+* `UserInterface#run()` calls `UserInterface#getConsoleCommand()` to read and parse the user's input.
+* `UserInterface#run()` will then call the corresponding command handler function based on the user's input.\
+In the example above, `ConsoleCommand` is an instance of `ConsoleCommandAddExpense` and hence,
+`UserInterface#runCommandAddExpense()` is called.
+* When the command handler function is called, it calls `Data` functions to perform operations on data.\
+In the example above, `UserInterface#runCommandAddExpense()` calls `Data#addExpense()` to add an expense to the program.
+* After the operations are performed, command handler functions calls `Storage` functions to save data.\
+In the example above, `UserInterface#runCommandAddExpense()` calls `Storage#saveToFile()` to save the newly added expense to a file.
+
+#### Component Interaction Sequence Diagrams:
+
+![Component-Interaction-On-Command-Entered](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/xzynos/tp/branch-MoneyGoWhere-Webpage/docs/diagrams/ComponentInteractionsRefGetUserCommand.puml)
+
+* `UserInterface#getConsoleCommand()` calls `UserInterface#getConsoleInput()` to read the user's input as a string.
+* `UserInterface#getConsoleCommand()` then calls `Parser#parse()` to parse the input string into the corresponding console command object.
+
+![Component-Interaction-On-Command-Entered](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/xzynos/tp/branch-MoneyGoWhere-Webpage/docs/diagrams/ComponentInteractionsRefPrintExpense.puml)
```

Comment:
This image reflected may not be intended for this section, do consider the image that has been added. 

![image](https://user-images.githubusercontent.com/88478793/198244156-cad0f33e-de6c-4888-a971-7418f796d654.png)